src/room.js

```diff
@@ -0,0 +1,40 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {
```

Comment:
Im not sure we need to add these new functions.
I think the update functions will do the job as they will create the property if its already exist.

all we need to do is use brackets as you said.

``` javascript 
  updateUsername (endpointId, username) {
    this.endpoints[endpointId][username] = username;
  }
```